### DISCUSSION TITLE
Reenable OpenWeather One Call API as mandatory API key

### DISCUSSION THREAD

**Discussion by dan9er** [breezy-weather/breezy-weather#997] [Ideas]
### Describe your idea

I have my OpenWeather API key that I use for the app. I was a bit disappointed to see that the weather information was degraded so heavily when I updated to v5.2.0.

Can the OneCall 3.0 option be reimplemented, but with the caveat that the user supplies their own API key?

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.1 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> # Features that will not be implemented
> Paid-only sources, too limited free-tier, or free-tier that requires privacy-invasive information (credit card info, phone number, etc)

OneCall API should not have been implemented initially, but back then it was one of the rare free and generous API, that decided later to make billing info mandatory.

Today, there are SO many free sources without one of these issues that I see no reason why I should give up my privacy or pay to maintain other sources for a few users.

Remember it's open source so you can implement any source you want and maintain it by yourself on your own fork, rebasing regularly and keeping up with the changes. I know a few users do. However, I think it will be best to look at other existing sources as they are some that perform better than OpenWeather that you could like.
